### PR TITLE
Gpio but on the internet

### PR DESCRIPTION
**Pull request by twoshark**
Endpoints for solenoid open, close, enable, disable
Handler Breakout
Mock Updates

### CODE REVIEW

File: src/io/rpi_pin.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func GetPins() []RpiPinMap {
 		RpiPinMap{
 			HeaderPin: 7,
 			BcmPin:    4,
-			Name:      "GPIO  7",
+			Name:      "GPIO 7",
```
Comment (dtp263) [inline src/io/rpi_pin.go]: nice

File: src/component/solenoid.go [hunk 2]
```diff
@@ -39,40 +52,53 @@ func (s *Solenoid) Disable() {
...
 
-func (s *Solenoid) open(duration int) {
-	if s.healthy() {
+//Open - Open the solenoid
+func (s *Solenoid) Open() {
+	if s.Healthy() {
+		log.Println(s.GPIO.Pin.Read(), s.State())
 		s.GPIO.Pin.High()
-		s.close(duration)
+		log.Println(s.GPIO.Pin.Read(), s.State())
```
Comment (dtp263) [inline src/component/solenoid.go]: extra log here?

File: src/cluster/cluster.go [hunk 3]
```diff
@@ -175,9 +229,9 @@ func (c *Cluster) JoinNetwork(URL string) error {
 
 //generateUniqueID returns a unique id for asigning to a new microcontroller
 func (c *Cluster) generateUniqueID() int {
-	randID := rand.Intn(100)
+	randID := rand.Intn(255)
```
Comment (dtp263) [inline src/cluster/cluster.go]: Could we add a a future TODO here for making this an ENV var or something more easily configurable.

File: src/app/routes.go [hunk 4]
```diff
@@ -26,9 +24,10 @@ func (a *Application) ConfigureRoutes(listenURL string) {
 
 	// Routes
 	a.Echo.GET("/", a.defaultGet)
-	a.Echo.POST("/", a.peerUpdate)
-	a.Echo.GET("/cluster_info", a.getClusterInfo)
-	a.Echo.POST("/join_network", a.joinNetwork)
+
+	a.addRegistrationRoutes()
+	a.addInfoRoutes()
+	a.addCommandRoutes()
```
[the remark this answers is not in the document]
Comment (dtp263) [reply src/app/routes.go]: or all in a folder called routes.

[the remark this answers is not in the document]
Comment (twoshark) [reply src/app/routes.go]: refactored the route pattern

File: src/cluster/cluster.go (hunk 3 above)
Comment (twoshark) [reply src/cluster/cluster.go]: This is done

**Pull request merged**